app/grandchallenge/notifications/models.py

```diff
@@ -1,27 +1,119 @@
-from actstream.models import Action
+from actstream.models import Action, followers
 from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.humanize.templatetags.humanize import naturaltime
 from django.db import models
+from django.utils.html import format_html
+from django.utils.translation import gettext_lazy as _
 from guardian.shortcuts import assign_perm
 
 from grandchallenge.core.models import UUIDModel
+from grandchallenge.profiles.templatetags.profiles import user_profile_link
+from grandchallenge.subdomains.utils import reverse
+
+
+class NotificationTypeChoices(models.TextChoices):
+    """Notification type choices."""
+
+    GENERIC = "GENERIC", _("Generic")
+    FORUM_POST = "FRM-POST", _("Forum post")
+    FORUM_POST_REPLY = "FRM-RPL", _("Forum post reply")
```

Comment:
Any reason to use abbreviations here, but full names later on (e.g. in `MISSING-METHOD`)?

Reply:
No good reason, I'll change them all to full names.